domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -949,9 +952,35 @@ public void saveBatch(String table, List<Object> entities, Handler<AsyncResult<R
       }
     } catch (Exception e) {
       replyHandler.handle(Future.failedFuture(e));
-      return;
+      return null;
+    }
+    return jsonArray;
+  }
+
+  /***
+   * Save a list of POJOs.
+   * POJOs are converted to json and saved in a single sql call. The generated IDs of the
+   * inserted records are returned in the ResultSet.
+   * @param table  destination table to insert into
+   * @param entities  each list element is a POJO
+   * @param replyHandler result, containing the id field for each inserted POJO
+   */
+  public void saveBatch(String table, List<Object> entities,
+    Handler<AsyncResult<ResultSet>> replyHandler) {
+
+    JsonArray jsonArray = batchEntitiesToJsonArray(entities, replyHandler);
+    if (jsonArray != null) {
+      saveBatch(table, jsonArray, replyHandler);
+    }
+  }
+
+  public void saveBatch(AsyncResult<SQLConnection> sqlConnection, String table,
+    List<Object> entities, Handler<AsyncResult<ResultSet>> replyHandler) {
+
+    JsonArray jsonArray = batchEntitiesToJsonArray(entities, replyHandler);
+    if (jsonArray != null) {
+      saveBatch(sqlConnection, table, jsonArray, DEFAULT_JSONB_FIELD_NAME, replyHandler);
     }
-    saveBatch(table, jsonArray, replyHandler);
   }
```

Comment:
Please add a unit test for jsonArray == null. After that please resubmit for code review!